src/jit/optcse.cpp

```diff
@@ -2070,15 +2107,23 @@ class CSE_Heuristic
                 cse->gtDebugFlags |= GTF_DEBUG_VAR_CSE_REF;
 #endif // DEBUG
 
-                // If we have side effects then we need to create a GT_COMMA tree instead
+                // Now we need to unmark any nested CSE's uses that are found in 'exp'
+                // As well as extract any nested CSE defs that are found in 'exp', these are appended to the sideEffList
+
+                // Afterwards the set of node in the 'sideEffectList' are preserved and
+                // all other nodes are removed and have their ref counts decremented
+                //
+                exp->gtCSEnum = NO_CSE; // clear the gtCSEnum field
```

Comment:
This cse field clear is just a small perf optimization, right? Or is it needed for correctness somehow?

Reply:
Well it has to be cleared now because optUnmarkCSEs will examine it.
And if it sees a CSE use it would preserved the whole tree.

Reply:
I'm not sure I understand -- we never keep the whole CSE use tree, just the side effecting and CSE def subtrees ... ?

Reply:
Yeah, You areright it is not a CSE def, but we would still to the work that we do for a CSE use, which is not what we want. So it is correct to clear the CSE use flag before calling optValnumCSE_UnmarkCSEs (and in turn UnmarkCSE)